src/winetricks

```diff
@@ -2546,10 +2555,10 @@ w_wine_version_in()
 # the environment variable WINETRICKS_BLACKLIST to disable it.
 w_workaround_wine_bug()
 {
-    if test "$WINE" = ""; then
-        echo "No need to work around wine bug $1 on Windows"
-        return 1
-    fi
```

Comment:
> A) fails Travis, so not going in

Resolved now and ready for review and merge

> B) I'm unclear on what this is needed for. I.e., what bug does this fix?

Referenced line was the main motivation for this commit since checking if WINE variable is zero seems unreliable and insane to me assuming that winetricks can be used on cygwin and WSL1/2 now

This way we have more control over logic per platform since we can reliably detect the platform in question and provide logic e.i: https://github.com/Winetricks/winetricks/pull/1318 which seems to require different workarounds for freebsd, darwin and linux. 

And assuming that these kernels are very different from each other i would expect using this logic to adapt wineapps for them and personally i'm trying to migrate from GNU/Linux on GNU/Redox which is going to need it to make WINE working.

Reply:
> Referenced line was the main motivation for this commit since checking if WINE variable is zero seems unreliable and insane to me assuming that winetricks can be used on cygwin and WSL1/2 now

"seems unreliable" isn't a bug. If you've got an actual bug, that's a different story. WINE is unset on windows, so that when a command line "$WINE" foo.exe is executed, it won't try to execute a non-existent binary.

I'm still not convinced winetricks actually works on WSL1/2 (even with this patch). I'd be happy to be proven wrong on that.

Reply:
@austin987 WSL1 has shell/bash running on what seems to be linux kernel so it's able to execute it for WSL2 don't know since it's not released yet from what i found i believe that WSL1 is something like WINE for linux and WSL2 is direct implementation of linux kernel.

> "seems unreliable" isn't a bug. If you've got an actual bug, that's a different story. 

```bash
load_something() { unset WINE ; w_workaround_wine_bug (windows is false triggered) ; }
```

Reply:
Yes, but it lacks X support. All it can run is wine cmd and other command line apps.

Reply:
@austin987 I was told that there are ways to get X support on the WSL1 from quick search: https://www.ctrl.blog/entry/how-to-x-on-wsl.html which seems to export Cygwin X/vcXsrv into WSL1 

![image](https://user-images.githubusercontent.com/11302521/66440235-aa6c4f00-ea32-11e9-909d-1555fee721f7.png)

Assuming that pty/tty is missing on WSL1 i would feel confident in making a franken-wsl and passing the X through SSH or something alike, but i'm not windows user so i won't be testing this.

Aldo there is probably no reason (excluding performance benefit?) to use WINE on windows so i would assume that WSL1 users would use it to install apps using CLI since winetricks has logic for them and is using AHK to make things faster to install.